BSP/Simulator_V2/Src/BSP_Timer.c

```diff
@@ -50,6 +59,6 @@ uint32_t BSP_Timer_GetRunFreq(void) {
  * @return  Microseconds 
  */
 uint32_t BSP_Timer_GetMicrosElapsed(void) {
-    // TODO
-    return 42;
+    return (uint32_t) (BSP_Timer_GetTicksElasped() * 1000000) / CLOCKS_PER_SEC;
```

Comment:
is uint32 big enough for this or should we bump it up to u64

Reply:
I think it should be big enough its like 4 billion for uint32. also Im not sure if i want to change to uint64 because functions should be the same as actual bsp layer